src/utils/metamaskUtils.ts

```diff
@@ -0,0 +1,18 @@
+const registerToken = async (earningTokenAddress: string, tokenName: string, tokenDecimals: number, image: string) => {
+  const tokenAdded = await (window as any).ethereum.request({
```

Comment:
Can the user add an asset without being connected on pancake ? If not, we should avoid using the window object to get the provider